setup.py

```diff
@@ -9,9 +9,13 @@
 # ray==2.5.0 broken: https://github.com/conda-forge/ray-packages-feedstock/issues/100
 # pydantic<2: https://github.com/modin-project/modin/issues/6336
 ray_deps = ["ray[default]>=1.13.0,!=2.5.0", "pyarrow>=7.0.0", "pydantic<2"]
-unidist_deps = ["unidist[mpi]>=0.2.1"]
+mpi_deps = ["unidist[mpi]>=0.2.1"]
 spreadsheet_deps = ["modin-spreadsheet>=0.1.0"]
-all_deps = dask_deps + ray_deps + unidist_deps + spreadsheet_deps
+# Currently, Modin does not include `mpi` option in `all`.
+# Otherwise, installation of modin[all] would fail because
+# users need to have a working MPI implementation and
+# certain software installed beforehand.
+all_deps = dask_deps + ray_deps + spreadsheet_deps
```

Comment:
I understand your motivation, but it confuses me that the target, which is called like `all`, is not that in essence.

Reply:
We have a similar issue with hdk. I don't think we want users to see the installation error with new releases of modin[all] for now. Maybe we need some time for users to understand the behavior/logic of modin[mpi].

Reply:
can we add a logic, checking if any supported `mpi` version is installed and then either install `unidist` or raise a warning about skipping it?

Reply:
Just checking if mpi is available on the system is not enough. There are much more other stuff that is important and required for mpi4py to be installed. mpi4py itself has the appropriate logic for observing an MPI library/compiler/etc. and I believe this should belong to mpi4py. To me it is fine to go with modin[mpi] for now.